scripts: ci: check_compliance: Add support for module Kconfigs

nordicjm opened the pull request:
Adds support for checking module samples and tests for additional Kconfigs, as well as logging Kconfigs, so that this check can be reused more easily with out of tree manifest repos

nordicjm committed:
scripts: ci: check_compliance: Add support for module Kconfigs

Adds support for checking module samples and tests for additional
Kconfigs, as well as logging Kconfigs, so that this check can be
reused more easily with out of tree manifest repos

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

pdgendt commented on scripts/ci/check_compliance.py:
```diff
@@ -723,6 +723,30 @@ def parse_kconfig(self):
+                continue
+
+            module_path = PurePath(project.abspath)
+            module_samples = module_path.joinpath('samples')
```
Tests and samples are properties of the module yaml files. Shouldn't we take those into account?

nordicjm replied:
They are? This is news to me... I will have a look

nordicjm committed:
zephyr: Add zephyr module file

Adds a module file which specifies the path to samples and tests
inside of zephyr

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented on scripts/ci/check_compliance.py:
```diff
@@ -723,6 +723,45 @@ def parse_kconfig(self):
+                module_yml = module_path.joinpath('zephyr/module.yaml')
+
+            if Path(module_yml).is_file():
+                with Path(module_yml).open('r', encoding='utf-8') as f:
```
open is the default but utf-8 is not and this is used elsewhere when reading in the module yaml files

nordicjm commented on scripts/ci/check_compliance.py:
```diff
@@ -723,6 +723,45 @@ def parse_kconfig(self):
+            if Path(module_yml).is_file():
+                with Path(module_yml).open('r', encoding='utf-8') as f:
+                    meta = yaml.load(f.read(), Loader=SafeLoader)
+                    dirs = []
```
Fixed
